cassandra/pool.py

```diff
@@ -363,20 +363,34 @@ def __init__(self, host, host_distance, session):
             return
 
         log.debug("Initializing connection for host %s", self.host)
-        self._connections[0] = session.cluster.connection_factory(host.endpoint)
+        first_connection = session.cluster.connection_factory(host.endpoint)
+        log.debug("first connection created for shard_id=%i", first_connection.shard_id)
+        self._connections[first_connection.shard_id] = first_connection
+        self.host.sharding_info = weakref.proxy(first_connection.sharding_info)
+
         self._keyspace = session.keyspace
+
         if self._keyspace:
-            self._connections[0].set_keyspace_blocking(self._keyspace)
+            first_connection.set_keyspace_blocking(self._keyspace)
+
+        for _ in range(first_connection.sharding_info.shards_count * 2):
+            conn = self._session.cluster.connection_factory(self.host.endpoint)
+            if conn.shard_id not in self._connections.keys():
+                log.debug("new connection created for shard_id=%i", conn.shard_id)
+                self._connections[conn.shard_id] = conn
+                if self._keyspace:
+                    self._connections[conn.shard_id].set_keyspace_blocking(self._keyspace)
+
+            if len(self._connections.keys()) == first_connection.sharding_info.shards_count:
+                break
+
         log.debug("Finished initializing connection for host %s", self.host)
```

Comment:
Don't you think that we'd rather do some sanity check here as to having a connection per shard?